executor/chroot/chroot_executor.go

```diff
@@ -0,0 +1,186 @@
+package chroot
+
+import (
+	"bytes"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"syscall"
+
+	"github.com/spacemonkeygo/errors"
+	"github.com/spacemonkeygo/errors/try"
+	"polydawn.net/repeatr/def"
+	"polydawn.net/repeatr/input"
+	"polydawn.net/repeatr/input/dispatch"
+	"polydawn.net/repeatr/lib/guid"
+)
+
+type Executor struct {
+	workspacePath string // default: something like '/var/lib/repeatr/executors/chroot/'.
+}
+
+func (x *Executor) Run(formula def.Formula) (job def.Job, outs []def.Output) {
+	try.Do(func() {
+		job, outs = x.run(formula)
+	}).Catch(input.InputError, func(e *errors.Error) {
+		// REVIEW: also directly pass input/output system errors up?  or, since we may have to gather several, put them in a group and wrap them in a "prereqs failed" executor error?
```

Comment:
I'm content to let this sit until we have more opinions about logging and production usage.